scripts.py

```diff
@@ -1381,22 +1406,26 @@ def load_collection(self, collection_name, data_source_name):
             )
 
             try:
-                mirror_integration = MirrorUploader.sitewide_integration(
-                    self._db
+                mirror_integration = MirrorUploader.integration_by_name(
+                    self._db, storage_name
                 )
             except CannotLoadConfiguration, e:
-                # There is no sitewide mirror configuration, or else
-                # there is more than one. Either way, we can't
-                # associate a mirror integration with the new collection.
+                # We can't associate any mirror integration with the new collection.
                 mirror_integration = None
 
             if mirror_integration:
                 collection.mirror_integration = mirror_integration
                 self.log.info(
-                    "Associated Collection %s with the sitewide storage integration.",
-                    collection.name
+                    "Associated Collection %s with the storage integration %s.",
+                    collection.name,
+                    storage_name
                 )
-        mirror = MirrorUploader.for_collection(collection)
+        mirror = dict(
+            covers=MirrorUploader.for_collection(
+                collection, ExternalIntegrationLink.COVERS),
+            books=MirrorUploader.for_collection(
+                collection, ExternalIntegrationLink.BOOKS)
+        )
```

Comment:
This is good, but after looking at the above code I think it doesn't go far enough. We can't complete the job of this script if there's nowhere to put the books, and if there's nowhere to put the covers we can only complete the job if it turns out there are no images. This would be a good time to raise an exception either MirrorUploader ends up None.